static/app/views/settings/project/projectFilters/projectFiltersSettings.tsx

```diff
@@ -189,48 +259,65 @@ class LegacyBrowserFilterRow extends Component<RowProps, RowState> {
     return (
       <div>
         {!disabled && (
-          <BulkFilter>
-            <BulkFilterLabel>{t('Filter')}:</BulkFilterLabel>
-            <ButtonBar gap={1}>
-              <Button
-                priority="link"
-                borderless
-                onClick={this.handleToggleSubfilters.bind(this, true)}
-              >
-                {t('All')}
-              </Button>
-              <Button
-                priority="link"
-                borderless
-                onClick={this.handleToggleSubfilters.bind(this, false)}
-              >
-                {t('None')}
-              </Button>
-            </ButtonBar>
-          </BulkFilter>
+          <div>
+            <BulkFilter>
+              <FieldLabel>{t('Legacy Browser Filters')}:</FieldLabel>
+              <ButtonBar gap={1}>
+                <Button
+                  priority="link"
+                  borderless
+                  onClick={this.handleToggleSubfilters.bind(this, true)}
+                >
+                  {t('All')}
+                </Button>
+                <Button
+                  priority="link"
+                  borderless
+                  onClick={this.handleToggleSubfilters.bind(this, false)}
+                >
+                  {t('None')}
+                </Button>
+              </ButtonBar>
+            </BulkFilter>
+            <FieldHelp>
+              {t(
+                'The browser versions filtered out will be periodically evaluated and updated'
+              )}
+            </FieldHelp>
+          </div>
         )}
 
         <FilterGrid>
-          {LEGACY_BROWSER_KEYS.map(key => {
-            const subfilter = LEGACY_BROWSER_SUBFILTERS[key];
-            return (
-              <FilterGridItem key={key}>
-                <FilterGridIcon src={subfilter.icon} />
-                <div>
-                  <FilterTitle>{subfilter.title}</FilterTitle>
-                  <FilterDescription>{subfilter.helpText}</FilterDescription>
-                </div>
-                <Switch
-                  aria-label={`${subfilter.title} ${subfilter.helpText}`}
-                  isActive={this.state.subfilters.has(key)}
-                  isDisabled={disabled}
-                  css={{flexShrink: 0, marginLeft: 6}}
-                  toggle={this.handleToggleSubfilters.bind(this, key)}
-                  size="lg"
-                />
-              </FilterGridItem>
-            );
-          })}
+          {Object.keys(LEGACY_BROWSER_SUBFILTERS)
+            .filter(key => {
+              if (this.props.hasLegacyBrowserUpdate) {
+                if (!LEGACY_BROWSER_SUBFILTERS[key].legacy) {
+                  return true;
+                }
+                return [...this.state.subfilters].includes(key);
```

Comment:
use the set functions if possible
```suggestion
                return this.state.subfilters.has(key);
```